src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -593,6 +740,71 @@ public function addAllowedTypes($option, $allowedTypes)
     }
 
     /**
+     * Adds allowed types for one or more options.
+     *
+     * If a nested option name is passed it will apply to all nested options.
+     *
+     * Any type for which a corresponding is_<type>() function exists is
+     * acceptable. Additionally, fully-qualified class or interface names may
+     * be passed.
+     *
+     * @param string|string[]|int $optionNames  One or more option names
+     *                                          or Options::ALL
+     * @param string|string[]     $allowedTypes One or more accepted types
+     * @param bool                $replace      Whether to replace previous
+     *                                          value
+     * @param int                 $nested       Whether this method is recursive
+     *                                          for nested options which is the
+     *                                          default. Pass Options::NONE to
+     *                                          change it.
+     *
+     * @return OptionsResolver This instance
+     *
+     * @throws UndefinedOptionsException If an option is undefined
+     * @throws AccessException           If called from a lazy option or normalizer
+     */
+    public function addAllowedTypesForAll($optionNames, $allowedTypes, $replace = false, $nested = Options::ALL)
```

Comment:
Should I do the same with `allowedValuesForAll()` ? There might be some use cases.